### ISSUE TITLE
Plugin uninstallable on Eclipse 2020-03

### ISSUE THREAD

**Issue by DavidGoodenough** [scalameta/metals-eclipse#9]
If you try to upgrade an existing Eclipse with the plugin installed, the upgrade process insists on removing the plugin, and if you try to install it on an Eclipse 2020-03 system it refused.  Both are due to dependency conflicts.

**Comment by ckipp01** [comment]
Hi @DavidDudson, thanks for the report! Yes, the eclipse plugin is a little behind. When one of the maintainers has a bit of time we'll tackle this. However, I can't promise on how soon it will be since none of us personally use the eclipse plugin.

**Comment by DavidGoodenough** [comment]
Not quite sure where you got the handle from, my name is David Goodenough not David Dudson. 

Anyway, I am happy to help if you can point me at development documentation.  I use Eclipse regularly (I have been using it since it was called VisualAge for Java - so a long time).  

I think it is just a matter of getting the dependencies right, although I understand that there were some API changes with the current LSP4J version which might need some code changes.  But those should be the same as for other LSP platforms.

**Comment by DavidGoodenough** [comment]
What are the current versions of the Metals things that this should depend on?

**Comment by ckipp01** [comment]
Hey, sorry for some reason, the only autocomplete option that came up was the wrong David, but I assumed it was right 😬 .

Regarding development documentation, we probably don't have much on the plugin itself, although @tgodzik will know more than me. Regarding the version of metals, it should be pointed towards the last stable release, which was 0.8.3.

Any help would be greatly appreciated.

**Comment by mundacho** [comment]
Hello, I coded most of the eclipse plugin. I'm currently very busy to update it :/. 

For the metals version, it's updated automatically to the latest if I remember well. However the problem, as you say is the lsp4e dependency which depends on lsp4j. I've both as dependencies in the project.

So the things to do are the following:

- update the lsp4e and lsp4j to the latest one
- fix errors that appear

If you have questions @DavidGoodenough I'll be happy to help. Hopefully next week I'll have some extra time.

Best regards,

Edmundo López Bóbeda

**Comment by DavidGoodenough** [comment]
@mundacho Did you do the development within Eclipse, and in particular its Plugin Development environment, and if so do you recall if it matters which version I use?

**Comment by PyvesB** [comment]
Hi there! I'm the author of the LSP-based Ruby Eclipse integration, I'm guessing the development environment should be fairly similar. [These steps](https://github.com/PyvesB/eclipse-solargraph#wrench-setting-your-own-working-copy-of-the-project) may help you get started. 😉

**Comment by mundacho** [comment]
I used the Eclipse IDE for Java Developers 2019-12. The plugin is written in scala, so you need to install the metals lsp plugin, and that's all. Some extra documentation is in the README file at https://github.com/scalameta/metals-eclipse/

**Comment by DavidGoodenough** [comment]
OK, I have a functioning 2019-12 with the plugin.  So I will give it a go.

**Comment by mundacho** [comment]
Hello @DavidGoodenough, any luck with that? I'm having some extra time today to take a look at it, but if you've already done it, I'll look into something else.

**Comment by DavidGoodenough** [comment]
I got stuck trying to get Eclipse to recognise the project (or should that be projects).  So no I have not got anywhere.  If you have time to add a little documentation as to how to load the clones repository into an Eclipse workspace I would be happy to help next time.  This time I suspect you will get there quicker.

**Comment by DavidGoodenough** [comment]
@mundacho I really have gotten nowhere with this.  Any chance you can either add some docs or do the upgrade?

**Comment by mundacho** [comment]
Hello, my latest attempt was stopped by some extra work that appeared out of nowhere. I'll give it a go this week. Sorry for the delay.

**Comment by mundacho** [comment]
Committed and deployed 👍

**Comment by DavidGoodenough** [comment]
Thank you I will give it a go

**Comment by DavidGoodenough** [comment]
@mundacho  For record, installed successfully.  Once again thank you.
